charts/nginx-ingress/values.yaml

```diff
@@ -167,6 +167,26 @@ controller:
   #   cpu: 1
   #   memory: 1Gi
 
+  ## The security context for the Ingress Controller pods.
+  podSecurityContext:
+    seccompProfile:
+      type: RuntimeDefault
+
+  ## The security context for the Ingress Controller containers.
+  containerSecurityContext: {} # Remove curly brackets before adding values
+    # allowPrivilegeEscalation: true
+    # readOnlyRootFilesystem: true # make sure this value is same as values.controller.readOnlyRootFilesystem
```

Comment:
if `values.controller.readOnlyRootFilesystem` is false and `controller.securityContext.readOnlyRootFilesystem` is true then:
```
F0213 17:24:18.939841       1 manager.go:151] Failed to write main config: failed to open /etc/nginx/nginx.conf: open /etc/nginx/nginx.conf: read-only file system
```
because init container won't create directories due to direct dependency on ``values.controller.readOnlyRootFilesystem` in templates

Reply:
If we're going to deprecate the original flag we need to fully support not using it, so we could add a helper function like:

```
{{- define "nginx-ingress.readOnlyRootFilesystem" -}}
{{- if or .Values.controller.readOnlyRootFilesystem (and .Values.controller.containerSecurityContext .Values.controller.containerSecurityContext.readOnlyRootFilesystem) -}}
true
{{- else -}}
false
{{- end -}}
{{- end -}}
```

and then where we check for it in the templates:

```
{{- if or ( eq (include "nginx-ingress.readOnlyRootFilesystem" .) "true" ) .Values.controller.initContainers }}
      initContainers:
{{- end }}
...
```